modules/tensor_mechanics/src/scalarkernels/GlobalStrain.C

```diff
@@ -37,14 +42,18 @@ GlobalStrain::GlobalStrain(const InputParameters & parameters)
     _pst_residual(_pst.getResidual()),
     _pst_jacobian(_pst.getJacobian()),
     _components(_var.order()),
-    _dim(_mesh.dimension())
+    _dim(_mesh.dimension()),
+    _periodic_dirs(getParam<MultiMooseEnum>("periodic_directions"))
```

Comment:
This doesn't seem like the right idea. You are asking the user to specify periodic bcs in two places. We should be querying periodic information either from the mesh (ideally), or the equation systems (where periodic bcs) are currently stored because they are enforced as constraints.

Reply:
My suggestion: query the periodic direction for all displacement variables. If they are not consistent periodicitywise throw an error. Otherwise use that as the periodicity info. (this will also simplify the thing I commented on where you use a TypeTensor<bool>. This will now be a TypeVector<bool> because all displacement variables should have the same periodicity. Otherwise none of this makes sense.

Reply:
Additionally, this object is only interested in periodic directions for displacement variables. If there are multiple periodic BC specified, fetching just the direction from mesh might provide incorrect information to the kernel.

Reply:
The reason why I said "ideally" the mesh is because we don't actually have complete periodic information on the mesh even though that seems like the logical place for it. If a user uses a translation function or vector, all of the mesh query stuff just won't work! That's because MOOSE currently doesn't store the periodic bcs anywhere, we just pass them through to libMesh:
https://github.com/idaholab/moose/blob/devel/framework/src/actions/AddPeriodicBCAction.C#L171

This _really_ should be fixed. We should add better query information to the mesh, or nonlinear system. All of the nicer functions on the mesh that work with `auto_direction` were added because we needed them and added them as we went.